nowriter/tmencoding/tm_encoder_bytesout_builder_adaptor.go

```diff
@@ -0,0 +1,94 @@
+package tmencoding
+
+import "bytes"
+import "time"
+
+type TMEncoderBytesOutBuilderAdaptor struct {
```

Comment:
@ethanfrey wait. if i forget to implement a method, the compiler will not allow the code to compile because the check would fail. so i don't think your initial narrative is accurate.

Reply:
huh well i removed the line you said and it still works so i must have misunderstood something. pushed the new code with that line deleted. thanks i guess. i will have to rethink this part a couple more times to be sure i didn't really want it. @ethanfrey

Reply:
The embedded interface means you accept a different struct that implements that interface and by default blindly relay any methods that match the interface to that struct.

Since you never refer to some other struct and just want to declare that your struct implements the interface, I made the suggestion to change

Reply:
now i get it thanks @ethanfrey